README
$Id: README,v 1.1.1.1 2010/04/15 09:55:56 peimic.comprock Exp $

Copyright (C) 2002 Michael Cannon <michael@peimic.com>
See full GNU Lesser General Public License in LICENSE.

This library is free software; you can redistribute it and/or
modify it under the terms of the GNU Lesser General Public
License as published by the Free Software Foundation; either
version 2.1 of the License, or (at your option) any later version.

This library is distributed in the hope that it will be useful,
but WITHOUT ANY WARRANTY; without even the implied warranty of
MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
Lesser General Public License for more details.

You should have received a copy of the GNU Lesser General Public
License along with this library; if not, write to the Free Software
Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA 02111-1307 USA



GENERAL:
cb_common abstracts the common module functions out of individual modules.
In doing this, I've created a module commonality area. Use phpMyEdit to create 
forms and their database handling near automatically. Review the template module
for examples.

Upon updates and installs, cb_common is designed to overwrite itself everytime.
That way previous release errors are minimized. If you customize cb_common
though, then you'll lose your edits so be sure to backup those changes. Even
better submit them to cbcvs@peimic.com for potential inclusion into the library.

Generally, Cannon BOSE module options are stored in a database. The tool used
for simplifying those operations is called phpMyEdit. Some features need
explanation of that tool:
Add - insert a new data row
More - insert a new data row and return to the Add screen
Change - modify a data row
View - display a data row's contents
Prev/Next - data row navigation buttons. They are disabled if not used for a
particular data view.


WARNING:
Please don't delete the cb_common directory or else Cannon BOSE modules will fail to work.


REGARDS:
Thank you for using a Cannon BOSE module. Please send questions, comments, and
concerns to cbcvs@peimic.com.

Michael 


ChangeLog
2002-11-25 23:02  comprock

	* class.CB_Block_Admin.php (1.21): Fix Call-time pass by refrence
	errors

2002-11-23 17:45  comprock

	* phpws_modules.php (1.2): New version

2002-11-22 22:30  comprock

	* ChangeLog (1.21), phpMyEdit/phpMyEdit.class.php (1.11): Update
	ChangeLog

2002-11-22 19:25  comprock

	* phpMyEdit/: phpMyEdit.class.php (1.10), phpMyEditSetup.php (1.4):
	Option fixes Warning and Notice fixes

2002-11-22 10:26  comprock

	* ChangeLog (1.20), debug_msg.php (1.33): Update ChangeLog

2002-11-22 09:06  comprock

	* ChangeLog (1.19): Update ChangeLog

2002-11-19 20:44  comprock

	* ChangeLog (1.18): Update ChangeLog

2002-11-19 20:09  comprock

	* class.CB_Block_Admin.php (1.20): Fix ALTER TABLE handling

2002-11-17 20:24  comprock

	* ChangeLog (1.17, rel_1-19-0): Update ChangeLog

2002-11-16 20:31  comprock

	* class.CB_Block_Admin.php (1.19, rel_1-19-0): File header fix

2002-11-16 19:19  comprock

	* class.CB_Block_Admin.php (1.18), beautify/beautify_php.class.inc
	(1.3, rel_1-19-0), beautify/beautify_php.php (1.3, rel_1-19-0):
	Verbiage fixes File write beautified code, beware of class
	duplication Directory handling fixes Menu adding and removal fixes

2002-11-16 14:43  comprock

	* README (1.28, rel_1-19-0), cb_database.php (1.34, rel_1-19-0),
	class.CB_Block_Admin.php (1.17): Fix sql_result2array() Add
	menu_remove() Add menu_post()

2002-11-16 14:09  comprock

	* cb_dir_file.php (1.30, rel_1-19-0): Fix virtual root to work with
	directories

2002-11-16 12:47  comprock

	* ChangeLog (1.16, rel_1-18-0): Update ChangeLog

2002-11-16 12:10  comprock

	* ChangeLog (1.15): Update ChangeLog

2002-11-16 12:07  comprock

	* class.CB_Block_Admin.php (1.16), phpws_modules.php (1.1,
	rel_1-19-0) (utags: rel_1-18-0): Use phpMyEdit to handle module
	block administration routines Add phpMyEdit configuration file for
	modules table

2002-11-15 14:03  comprock

	* INSTALL (1.26, rel_1-19-0, rel_1-18-0), LICENSE (1.24,
	rel_1-19-0, rel_1-18-0), README (1.27, rel_1-18-0), TODO (1.25,
	rel_1-19-0, rel_1-18-0), cb_array.php (1.30, rel_1-19-0,
	rel_1-18-0), cb_common.php (1.32, rel_1-19-0, rel_1-18-0),
	cb_common_alt.php (1.31, rel_1-19-0, rel_1-18-0), cb_database.php
	(1.33, rel_1-18-0), cb_date_time.php (1.31, rel_1-19-0,
	rel_1-18-0), cb_db_table_to_class.php (1.10, rel_1-19-0,
	rel_1-18-0), cb_dir_file.php (1.29, rel_1-18-0), cb_email.php
	(1.27, rel_1-19-0, rel_1-18-0), cb_html.php (1.31, rel_1-19-0,
	rel_1-18-0), cb_javascript.php (1.20, rel_1-19-0, rel_1-18-0),
	cb_math.php (1.27, rel_1-19-0, rel_1-18-0), cb_string.php (1.33,
	rel_1-19-0, rel_1-18-0), cb_validation.php (1.19, rel_1-19-0,
	rel_1-18-0), class.CB_Block_Admin.php (1.15), class.DB_Pager.php
	(1.5, rel_1-19-0, rel_1-18-0), debug_msg.php (1.32, rel_1-19-0,
	rel_1-18-0), index.php (1.19, rel_1-19-0, rel_1-18-0), phpinfo.php
	(1.27, rel_1-19-0, rel_1-18-0): IC to SF updates

2002-11-09 15:43  comprock

	* phpMyEdit/images/: pme-change.png (1.4), pme-copy.png (1.4),
	pme-delete.png (1.4), pme-view.png (1.4) (utags: rel_1-18-0,
	rel_1-19-0): Reload pics

2002-11-09 15:43  comprock

	* phpMyEdit/images/: pme-change.png (1.3), pme-copy.png (1.3),
	pme-delete.png (1.3), pme-view.png (1.3): Attempt to fix images

2002-11-09 14:35  comprock

	* phpMyEdit/: .cvsignore (1.1), Makefile (1.1), phpMyEditSetup.php
	(1.3), timer.class (1.2), comprehensive/.cvsignore (1.1),
	comprehensive/comprehensive.php (1.1),
	comprehensive/comprehensive.sql (1.1), doc/.cvsignore (1.1),
	doc/AUTHORS (1.2), doc/ChangeLog (1.3), doc/README (1.3), doc/TODO
	(1.2), doc/sgml/configuration.fields.sgml (1.1),
	doc/sgml/configuration.general.sgml (1.1),
	doc/sgml/configuration.sgml (1.1), doc/sgml/defs.sgml (1.1),
	doc/sgml/extensions.sgml (1.1), doc/sgml/installation.sgml (1.1),
	doc/sgml/introduction.sgml (1.1), doc/sgml/other-information.sgml
	(1.1), doc/sgml/phpMyEdit.sgml (1.1),
	doc/sgml/images/id-selection.gif (1.1), doc/sgml/images/logon.gif
	(1.1), doc/sgml/images/result-script.gif (1.1),
	doc/sgml/images/table-selection.gif (1.1),
	extensions/phpMyEdit-report.class.php (1.1),
	extensions/phpMyEdit-slide.class.php (1.1),
	images/gif/pme-change.gif (1.1), images/gif/pme-copy.gif (1.1),
	images/gif/pme-delete.gif (1.1), images/gif/pme-view.gif (1.1),
	images/ori/pme-change.gif (1.1), images/ori/pme-copy.gif (1.1),
	images/ori/pme-delete.gif (1.1), images/ori/pme-view.gif (1.1),
	lang/PME.lang.DE.inc (1.3), lang/PME.lang.EN-US.inc (1.3),
	lang/PME.lang.EN.inc (1.3), lang/PME.lang.FR.inc (1.3),
	lang/PME.lang.IT.inc (1.3), lang/PME.lang.NL.inc (1.3),
	lang/PME.lang.PG.inc (1.3), lang/PME.lang.PL.inc (1.2),
	lang/PME.lang.SK.inc (1.3), lang/PME.lang.SP.inc (1.3) (utags:
	rel_1-18-0, rel_1-19-0): phpMyEdit 5.0 to 5.1 update

2002-11-09 14:28  comprock

	* phpMyEdit/phpMyEdit.class.php (1.9, rel_1-19-0, rel_1-18-0): Fix
	$fdd value/description pairing

2002-11-09 13:37  comprock

	* phpMyEdit/phpMyEdit.class.php (1.8): Platon 1.26 update Combine
	with CB changes

2002-11-09 07:51  comprock

	* cb_dir_file.php (1.28): Add virtual_root()

2002-11-08 22:26  comprock

	* ChangeLog (1.14): Update ChangeLog

2002-11-08 10:54  comprock

	* INSTALL (1.25), LICENSE (1.23), README (1.26), TODO (1.24),
	cb_array.php (1.29), cb_common.php (1.31), cb_common_alt.php
	(1.30), cb_database.php (1.32), cb_date_time.php (1.30),
	cb_db_table_to_class.php (1.9), cb_dir_file.php (1.27),
	cb_email.php (1.26), cb_html.php (1.30), cb_javascript.php (1.19),
	cb_math.php (1.26), cb_string.php (1.32), class.CB_Block_Admin.php
	(1.14), index.php (1.18), phpinfo.php (1.26): IC to SF

2002-11-06 07:36  comprock

	* class.DB_Pager.php (1.4): Add missing variable Rename internal
	default variables

2002-11-04 15:30  comprock

	* debug_msg.php (1.31): Enable $show_globals for php_variables()

2002-11-04 14:46  comprock

	* cb_validation.php (1.18): Add is_blank()

2002-11-04 10:43  comprock

	* debug_msg.php (1.30): Add php_variables()

2002-11-04 09:56  comprock


cb_common.php
<?php

	/**
	 * Cannon BOSE's centralization of common functions
	 *
	 * Copyright (C) 2002 Michael Cannon <michael@peimic.com>
	 * See full GNU Lesser General Public License in LICENSE.
	 *
	 * @author Michael Cannon <michael@peimic.com>
	 * @package cb_common
	 * @version $Id: cb_common.php,v 1.1.1.1 2010/04/15 09:55:56 peimic.comprock Exp $
	 */


	// CB file's location
	$cb_root = dirname(__FILE__) . '/';
	$cb_root = windows_path_fix($cb_root);

	// module block and setup code
	include_once($cb_root . 'class.CB_Block_Admin.php');

	// database pager
	include_once($cb_root . 'class.DB_Pager.php');

	// array functions
	include_once($cb_root . 'cb_array.php');

	// database functions
	include_once($cb_root . 'cb_database.php');

	// date and time functions
	include_once($cb_root . 'cb_date_time.php');

	// debug tools
	include_once($cb_root . 'debug_msg.php');

	// file and directory functions
	include_once($cb_root . 'cb_dir_file.php');

	// file upload helper
	include_once($cb_root . 'fileupload/fileupload-class.php');

	// e-mail functions
	include_once($cb_root . 'cb_email.php');
	include_once($cb_root . 'htmlMimeMail/htmlMimeMail.php');

	// html functions
	include_once($cb_root . 'cb_html.php');

	// javascript functions
	include_once($cb_root . 'cb_javascript.php');

	// math functions
	include_once($cb_root . 'cb_math.php');

	// phpMyEdit functions
	include_once($cb_root . 'phpMyEdit/phpMyEdit.class.php');

	// string functions
	include_once($cb_root . 'cb_string.php');

	// validation functions
	include_once($cb_root . 'cb_validation.php');



	/**
	 * Fix $path to work on windows machines.
	 *
	 * @param $path URI from PHP function, ex: getcwd(), 'c:/documents/files/'
	 * @return string
	 */
	function windows_path_fix($path)
	{
		// look at $_ENV['OS']
		// is this running on a Windows machine
		if ( isset($_ENV['OS']) && preg_match('/window/i', $_ENV['OS']) )
		{
			// if Windows...
			// convert forward slashes '/' to back slashes '\'
			$path = preg_replace('/\//', '\\', $path);
		}

		return $path;
	}

?>

